### PR TITLE
added z-jump check

### PR DESCRIPTION
**Pull request by mgeplf**
* Follows BBP policy of what a z-jump is:
    Large movements in the Z axis between two points

### CODE REVIEW

**Comment by mgeplf** [comment]
need

File: neurom/check/neuron_checks.py [hunk 1]
```diff
@@ -171,3 +171,25 @@ def has_nonzero_soma_radius(neuron, threshold=0.0):
...
+        neuron: Neuron object whose neurites will be tested
+        max_z_distance: value above which consecutive z-values are
+        considered a jump
+
+    Return:
+        status and list of ids bad sections
+    '''
+    bad_ids = []
+    for s in _nf.iter_sections(neuron):
+        it = izip(s.points, s.points[1:])
```
Comment (lidakanari) [inline neurom/check/neuron_checks.py]: Thanks Mike! Do we need to detect jumps between any consecutive points of beginning and end of a segment? I think it should be the second, which from my understanding is not what is done here, but my interpretation of this error is a bit blurry.

Comment (mgeplf) [inline neurom/check/neuron_checks.py]: Isn't a segment defined by two points?

Comment (lidakanari) [inline neurom/check/neuron_checks.py]: Now I see the difference: those are two points within a section, right? It's OK then, I'm saying the same thing.

File: apps/morph_check [hunk 2]
```diff
@@ -84,13 +84,15 @@ Default errors checked for
...
 * Zero length sections
+* Jumps in the Z-direction
 
 Default values for options
 --------------
 * has_nonzero_soma_radius 0.0
 * has_all_nonzero_neurite_radii: 0.007
 * has_all_nonzero_segment_lengths: 0.01
 * has_all_nonzero_section_lengths: 0.01
+* has_no_zjumps: 30 micrometer - distance considered a jump
```
Comment (juanchopanza) [inline apps/morph_check]: Do we really want this switched on by default? Especially given that it arbitrarily picks one axis, so is not very general. We could let users enable this in their config file.

Comment (mgeplf) [inline apps/morph_check]: As we discussed, my plan is to:
1) Turn this off by default
2) allow one to pick the 'axis' - not everyone does reconstructions with slice in z axis, however, it should always (?) be one of x, y, z

Comment (juanchopanza) [inline apps/morph_check]: That sounds fine to me!

File: neurom/check/runner.py [hunk 3]
```diff
@@ -126,13 +126,14 @@ def _check_file(self, f):
...
         try:
             data = load_data(f)
-            result &= self._check_loop(data, 'structural_checks')
-            nrn = fst_core.FstNeuron(data)
-            result &= self._check_loop(nrn, 'neuron_checks')
         except StandardError as e:
             L.error('Failed to load data... skipping tests for this file')
             L.error(e.message)
-            result = False
+            return False, {f: OrderedDict(('ALL', False))}
```
Comment (mgeplf) [inline neurom/check/runner.py]: I wrote my config.yaml wrong, but it was error out saying it was b/c the file couldn't load, and that was confusing.

File: neurom/check/runner.py [hunk 4]
```diff
@@ -126,12 +126,17 @@ def _check_file(self, f):
...
+            return False, {f: OrderedDict([('ALL', False)])}
+
+        try:
             result &= self._check_loop(data, 'structural_checks')
             nrn = fst_core.FstNeuron(data)
             result &= self._check_loop(nrn, 'neuron_checks')
         except StandardError as e:
-            L.error('Failed to load data... skipping tests for this file')
-            L.error(e.message)
+            L.error('Check failed: %s: %s', type(e), e.message)
```
Comment (mgeplf) [inline neurom/check/runner.py]: not sure happy w/ this, but it's the previous behavior.

**Comment by mgeplf** [comment]
does this satisfy #272?

**Pull request merged**